qp_target_gene/__init__.py

```diff
@@ -5,3 +5,152 @@
+
+from qiita_client import QiitaPlugin, QiitaCommand
+
+from .split_libraries import split_libraries, split_libraries_fastq
+from .pick_otus import pick_closed_reference_otus
+
+# Initialize the plugin
+plugin = QiitaPlugin(
+    'QIIME', '1.9.1', 'Quantitative Insights Into Microbial Ecology (QIIME) '
+                      'is an open-source bioinformatics pipeline for '
+                      'performing microbiome analysis from raw DNA '
+                      'sequencing data')
+
+# Define the Split libraries command
+req_params = {'input_data': ('artifact', ['FASTA' 'FASTA_Sanger', 'SFF'])}
+opt_params = {
+    'barcode_type': ['string', 'golay_12'],
+    'disable_bc_correction': ['bool', 'False'],
+    'disable_primers': ['bool', 'False'],
+    'max_ambig': ['integer', '6'],
+    'max_barcode_errors': ['float', '1.5'],
+    'max_homopolymer': ['integer', '6'],
+    'max_primer_mismatch': ['integer', '0'],
+    'max_seq_len': ['integer', '1000'],
+    'min_qual_score': ['integer', '25'],
+    'min_seq_len': ['integer', '200'],
+    'qual_score_window': ['integer', '0'],
+    'reverse_primer_mismatches': ['integer', '0'],
+    'reverse_primers': [
+        'choice:["disable", "truncate_only", "truncate_remove"]', 'disable'],
+    'trim_seq_length': ['bool', 'False'],
+    'truncate_ambi_bases': ['bool', 'False']}
+outputs = {'demultiplexed': 'Demultiplexed'}
+dflt_param_set = {
+    'Defaults with Golay 12 barcodes': {
+        'reverse_primers': 'disable',
+        'reverse_primer_mismatches': 0, 'disable_bc_correction': False,
+        'max_barcode_errors': 1.5, 'disable_primers': False,
+        'min_seq_len': 200, 'truncate_ambi_bases': False, 'max_ambig': 6,
+        'min_qual_score': 25, 'trim_seq_length': False, 'max_seq_len': 1000,
+        'max_primer_mismatch': 0, 'max_homopolymer': 6, 'qual_score_window': 0,
+        'barcode_type': 'golay_12'},
+    'Defaults with Hamming 8 barcodes': {
+        'reverse_primers': 'disable', 'reverse_primer_mismatches': 0,
+        'disable_bc_correction': False, 'max_barcode_errors': 1.5,
+        'disable_primers': False, 'min_seq_len': 200,
+        'truncate_ambi_bases': False, 'max_ambig': 6, 'min_qual_score': 25,
+        'trim_seq_length': False, 'max_seq_len': 1000,
+        'max_primer_mismatch': 0, 'max_homopolymer': 6, 'qual_score_window': 0,
+        'barcode_type': 'hamming_8'}}
+sl_cmd = QiitaCommand(
+    "Split libraries",
+    "Demultiplexes and applies quality control to FASTA data",
+    split_libraries, req_params, opt_params, outputs, dflt_param_set)
+plugin.register_command(sl_cmd)
+
+# Define the Split libraries FASTQ command
+req_params = {'input_data': ('artifact', ['FASTQ', 'per_sample_FASTQ'])}
+opt_params = {
+    'barcode_type': ['string', 'golay_12'],
+    'max_bad_run_length': ['integer', '3'],
+    'max_barcode_errors': ['float', '1.5'],
+    'min_per_read_length_fraction': ['float', '0.75'],
+    'phred_offset': ['string', ''],
```

Comment:
Changed to choice - this change is pairing https://github.com/biocore/qiita/pull/1963 which I  just noticed that have errors and I'm fixing right now.